packages/proxy/schema/index.test.ts

```diff
@@ -0,0 +1,152 @@
+      stream: true,
+    },
+    google: {
+      max_tokens: 1500,
+      messages: [
+        {
+          content: "You are a helpful assistant.",
+          role: "system",
+        },
+        {
+          content: "Hello, how are you?",
+          role: "user",
+        },
+      ],
+      model: "gpt-4o",
+      stream: true,
+      temperature: 0.7,
+      top_p: 0.9,
+    },
+    anthropic: {
+      max_tokens: 1500,
+      messages: [
+        {
+          content: "You are a helpful assistant.",
+          // @ts-expect-error -- TODO: shouldn't we have translated this to a non system role?
+          role: "system",
+        },
+        {
+          content: "Hello, how are you?",
+          role: "user",
+        },
+      ],
+      model: "gpt-4o",
+      stream: true,
+      temperature: 0.7,
+      top_p: 0.9,
+    },
+  },
+  reasoning: {
+    openai: {
+      model: "gpt-4o",
+      messages: [
+        {
+          role: "system",
+          content: "You are a detailed reasoning assistant.",
+        },
+        {
+          role: "user",
+          content: "Explain how to solve 2x + 4 = 12 step by step.",
+        },
+      ],
+      temperature: 0,
+      max_tokens: 1000,
+      reasoning_effort: "high",
+      stream: false,
+    },
+    google: {
+      model: "gpt-4o",
+      // notice how this is still an intermediate param
+      // google's api expects a content instead of messages, for example
+      messages: [
+        {
+          role: "system",
+          content: "You are a detailed reasoning assistant.",
+        },
+        {
+          role: "user",
+          content: "Explain how to solve 2x + 4 = 12 step by step.",
+        },
+      ],
+      temperature: 0,
+      thinkingConfig: {
+        thinkingBudget: 4096,
+        includeThoughts: true,
+      },
+      maxOutputTokens: 5120,
+      stream: false,
+    },
+    anthropic: {
+      model: "gpt-4o",
+      messages: [
+        {
+          // @ts-expect-error  -- we use the role to later manipulate the request
+          role: "system",
+          content: "You are a detailed reasoning assistant.",
+        },
+        {
+          role: "user",
+          content: "Explain how to solve 2x + 4 = 12 step by step.",
+        },
+      ],
+      max_tokens: 5120,
+      temperature: 1,
+      stream: false,
+      thinking: {
+        budget_tokens: 4096,
+        type: "enabled",
+      },
+    },
+  },
+};
+
+Object.entries(examples).forEach(([example, { openai, ...providers }]) => {
+  Object.entries(providers).forEach(([provider, expected]) => {
+    it(`[${example}] translate openai to ${provider} params`, () => {
+      const result = translateParams(
+        provider as ModelFormat,
+        openai as unknown as Record<string, unknown>,
+      );
+      try {
+        expect(result).toEqual(expected);
+      } catch (error) {
+        try {
+          // try to relax the output a little
+          expect(result).toMatchObject(expected);
```

Comment:
if you want to save some typing we can use `.toMatchInlineSnapshot` and sanity check each output before merging this PR. then subsequent changes can easily be seen in the snapshots

Reply:
Just tried this, and noticed we lost some type protections.